sql/2019/05_Third_Parties/05_03.sql

```diff
@@ -0,0 +1,19 @@
+#standardSQL
+# Percentage of requests that are third party requests broken down by third party category by resource type.
```

Comment:
@rviscomi for something like this, I was assuming the percentage is trivially computed by hand but based on https://github.com/HTTPArchive/almanac.httparchive.org/issues/67#issuecomment-517350413 it seems as though this sort of `GROUP BY` result won't be very well handled. Do I need to manually explode this set of options to arrive at one static query?

Reply:
Could you calculate the percent in the query? Something like this:

```sql
#standardSQL
# Percentage of requests that are third party requests broken down by third party category by resource type.
SELECT
  thirdPartyCategory,
  contentType,
  COUNT(*) AS totalRequests,
  ROUND(COUNT(0) * 100 / SUM(COUNT(0)) OVER (), 2) AS pctRequests
FROM (
  SELECT
      resp_content_type AS contentType,
      ThirdPartyTable.category as thirdPartyCategory
    FROM
      `httparchive.almanac.summary_requests`
    LEFT JOIN
      `lighthouse-infrastructure.third_party_web.2019_07_01` AS ThirdPartyTable
    ON NET.HOST(url) = ThirdPartyTable.domain
)
GROUP BY
  thirdPartyCategory,
  contentType
ORDER BY
  totalRequests DESC
```